bluesky/simulators.py

```diff
@@ -114,3 +121,401 @@ def check_limits(plan):
                 raise LimitsExceeded("This plan would put {} at {} "
                                      "which is outside of its limits, {}."
                                      "".format(msg.obj.name, pos, (low, high)))
+
+
+
+
+
+"""
+ATTRIBUTE PARAMETERS
+--------------------
+_run_start_cmds: list
+    A list of message 'commands' that are found at the start of a run.
+_run_end_cmds: list
+    A list of message 'commands' that are found at the end of a run.
+_group_start_cmds: list
+    A list of message 'commands' that are found at the begining of a 'group' (where more than
+    one command is processed in parallel).
+_group_end_cmds: list
+    A list of message 'commands' that are found at the begining of a 'group' (where more than
+    one command is processed in parallel).
+_self._timed_cmds: list.
+    A list of message 'commands' that do not start/end a run/group/flyer but are stil timed.
+__flyer_start_cmds: list.
+    A list of message 'commands' that start a 'flyer'.
+_plan_history: dict.
+    A dictionary containing information on values updated during the plan. It has key:arg 
+    pairs with keys relating to message components where each arg is a dictionary 
+    containing devices (detectors, axes) that have been updated, as keywords and a value 
+    indicating the update status. For 'set' the update status is the latest 'position' that
+    it is to be set and for 'trigger' the update status is the number of times since the 
+    beginning of the scan or since an 'unstage' event that the device has been triggered.
+
+"""
+#The commands that start/end a 'run'.
```

Comment:
Most -- maybe all -- of these lists strike me as unneeded obfuscation. There is only one command that starts are run, `'open_run'`, and it is not likely to change in the future. Why not just type `'open_run'` directly where it's needed?

It occurs to me you might intend this as path to registering custom user commands (message types) with this system. User commands are somewhat limited: a user can't add a comment that interacts with private RunEngine state, including opening/closing runs or doing anything with groups. It is safe to hard-code these command names, for the foreseeable future.

Reply:
I am not convinced that hard-coding these is a good idea. My strong preference is to leave it as is in case things change in the future to make it easier to change.